components/scream/src/physics/p3/p3_nr_conservation_impl.hpp

```diff
@@ -17,7 +17,7 @@ void Functions<S,D>::nr_conservation(const Spack& nr, const Spack& ni2nr_melt_te
 {
   const auto sink_nr = (nr_collect_tend + nr2ni_immers_freeze_tend + nr_selfcollect_tend + nr_evap_tend)*dt;
   const auto source_nr = nr + (ni2nr_melt_tend*nmltratio + nr_ice_shed_tend + ncshdc + nc2nr_autoconv_tend)*dt;
-  const auto mask = sink_nr > source_nr && context;
+  const auto mask = sink_nr > source_nr && context && sink_nr > 0;
```

Comment:
!= 0

Reply:
Same as sink_ni.

Reply:
@ambrad , this change does appear to be necessary. When I remove it, I get FPEs in all the p3_tests.